src/material-experimental/mdc-checkbox/checkbox.ts

```diff
@@ -228,12 +232,28 @@ export class MatCheckbox implements AfterViewInit, OnDestroy, ControlValueAccess
       private _changeDetectorRef: ChangeDetectorRef,
       private _platform: Platform,
       @Attribute('tabindex') tabIndex: string,
+      /**
+       * @deprecated `_clickAction` parameter to be removed, use
+       * `MAT_CHECKBOX_DEFAULT_OPTIONS`
+       * @breaking-change 10.0.0
+       */
       @Optional() @Inject(MAT_CHECKBOX_CLICK_ACTION) private _clickAction: MatCheckboxClickAction,
-      @Optional() @Inject(ANIMATION_MODULE_TYPE) public _animationMode?: string) {
-    this.tabIndex = parseInt(tabIndex) || 0;
-    this._checkboxFoundation = new MDCCheckboxFoundation(this._checkboxAdapter);
+      @Optional() @Inject(ANIMATION_MODULE_TYPE) public _animationMode?: string,
+      @Optional() @Inject(MAT_CHECKBOX_DEFAULT_OPTIONS)
+          private _options?: MatCheckboxDefaultOptions) {
     // Note: We don't need to set up the MDCFormFieldFoundation. Its only purpose is to manage the
     // ripple, which we do ourselves instead.
+    this.tabIndex = parseInt(tabIndex) || 0;
+    this._checkboxFoundation = new MDCCheckboxFoundation(this._checkboxAdapter);
+
+    this._options = this._options || {};
```

Comment:
You're only using the options in the constructor so you don't need to create a property for them.

Reply:
True, though it does make it a little nicer for the following lines where I directly access `this._options` instead of adding an `if(this._options) { ... }` check before them.

Reply:
Can we use default parameters on constructors that get DI'd? i.e.:
`@Optional() @Inject(MAT_CHECKBOX_DEFAULT_OPTIONS) private _options: MatCheckboxDefaultOptions = {}`

Reply:
Unfortunately not, Angular will inject `null` if it's not provided: https://stackblitz.com/edit/angular-euxmcx?file=src/app/app.component.ts